linux/LibSensors.c

```diff
@@ -261,6 +261,43 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
       goto out;
    }
 
+   /* Read sysfs to get HT/SMT cores and correctly distribute the temperature values */
+   if (coreTempCount < activeCPUs) {
+      double oldData[existingCPUs + 1];
```

Comment:
Variable length array is not recommended in C.

Reply:
Please avoid VLA. Although they are part of C99 (which htop targets), there are later language versions that make them optional. Also, having dynamically sized data on the stack is an accident waiting to happen, from a security PoV. Thus: No VLAs!

Reply:
Sorry for a late response, but isn't this also a VLA?
https://github.com/htop-dev/htop/blob/477c20bf7b7c40b84009cd221669e990ad6562a6/linux/LibSensors.c#L146
That's where I got this from.

Reply:
The LibSensors code is not the cleanest in some places … ;-)

That said, I did some cleanup in #1273.